chapters/vertex_input_data_processing.md

```diff
@@ -255,6 +232,62 @@ const VkVertexInputAttributeDescription attributes[] = {
 
 ![vertex_input_data_processing_example_d_vertex](../images/vertex_input_data_processing_example_d_vertex.png)
 
+## Example E - understanding input attribute format
+
+The `VkVertexInputAttributeDescription::format` can be the cause of confusion. The `format` field just describes the **size** and **type** of the data the shader should read in.
+
+The reason for using the `VkFormat` values is they are well defined and match the input layouts of the vertex shader.
+
+For this example the vertex data is just four floats:
+
+```c
+struct Vertex {
+    float a, b, c, d;
+};
+```
+
+The data being read will be overlapped from how the `format` and `offset` is set
+
+```c
+const VkVertexInputBindingDescription binding = {
+    0,                          // binding
+    sizeof(Vertex),             // stride
+    VK_VERTEX_INPUT_RATE_VERTEX // inputRate
+};
+
+const VkVertexInputAttributeDescription attributes[] = {
+    {
+        0,                          // location
+        binding.binding,            // binding
+        VK_FORMAT_R32G32_SFLOAT,    // format - Reads in two 32-bit signed floats
+        0                           // offset
+    },
+    {
+        1,                          // location
+        binding.binding,            // binding
+        VK_FORMAT_R32G32B32_SFLOAT, // format - Reads in three 32-bit signed floats
```

Comment:
Same here `// format - Reads in three 32-bit signed floats` becomes `// format - Points at three 32-bit signed floats "b", "c" and "d"` providing a hint that the last component will be discarded later.

Reply:
```suggestion
        VK_FORMAT_R32G32B32_SFLOAT, // format - Reads in three 32-bit signed floats ('b', 'c', and 'd')
```